src/category_theory/limits/opposites.lean

```diff
@@ -0,0 +1,76 @@
+-- Copyright (c) 2019 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Scott Morrison
+
+import category_theory.limits.limits
+
+universes v u
+
+open category_theory
+open category_theory.functor
+
+namespace category_theory.limits
+
+variables {C : Sort u} [𝒞 : category.{v+1} C]
+include 𝒞
+variables {J : Type v} [small_category J]
+variable (F : J ⥤ Cᵒᵖ)
+
+instance [has_colimit.{v} F.left_op] : has_limit.{v} F :=
+{ cone := cone_of_cocone_left_op (colimit.cocone F.left_op),
+  is_limit :=
+  { lift := λ s, (colimit.desc F.left_op (cocone_left_op_of_cone s)).op,
+    fac' := λ s j,
+    begin
+      rw [cone_of_cocone_left_op_π_app, colimit.cocone_ι, ←op_comp,
+          colimit.ι_desc, cocone_left_op_of_cone_ι_app, has_hom.hom.op_unop],
+      refl, end,
+    uniq' := λ s m w,
+    begin
+      -- It's a pity we can't do this automatically.
+      -- Usually something like this would work by limit.hom_ext,
+      -- but the opposites get in the way of this firing.
+      have u := (colimit.is_colimit F.left_op).uniq (cocone_left_op_of_cone s) (m.unop),
```

Comment:
Can you robustify this proof by replacing `dsimp` with `change foobar` and `simp` with the output of `squeeze_simp` (or maybe it's a `simpa`)?

Reply:
Fixed.